In [1]:
# always run
from random import randint
import json, stanza

# Data Wrangling

In [2]:
x = 0
file = json.load(open(f'database_dump_drugs/{x}.json'))
print(file[0]['article_content'])

FileNotFoundError: [Errno 2] No such file or directory: 'database_dump_drugs/0.json'

In [ ]:
# proof that article_content == article_contentRaw

duplicates = 0
aes = []

for x in range(100):
    file = open(f'database_dump_drugs/{x}.json',)
    articles = json.load(file)
    if x % 100 == 0:
        print(f'At json file {x} of 1571')
    for y in range(len(articles)):
        if articles[y]['article_content'] in aes:
            duplicates += 1
            print(f'Hit at json file {x} in article {y}')
        aes.append(articles[y]['article_content'])

print(duplicates, len(aes))

In [ ]:
print(duplicates, len(aes))

35609 50000


So we have 35609 unique articles in the dataset out of the 50000 original ones. 

In [ ]:
for x in range(10):
    js = randint(0, 1571)
    file = json.load(open(f'database_dump_drugs/{x}.json'))
    art = randint(0, len(file))
    print(file[art]['article_content'])
    print("")
    print("")
    print("")

Glasgow drugs recovery linked to Orkney Drugs Dog support
October 23, 2020 at 10:16 am
POLICE say that a 24-year-old woman has been charged following the recovery of Class A drugs in Glasgow, linked to intelligence received by officers in Orkney.
Local officers from the Kirkwall Community Policing Team launched an investigation into suspected drug offences and were assisted by the charity Orkney Drugs Dog, as well as Police Scotland colleagues in Greater Glasgow Police Division.
A warrant was executed in the Possilpark area of the city on Monday, October 19, and a quantity of drugs with an estimated ‘street’ value in excess of £10,000 were recovered. The woman, from Glasgow, has been reported to the Procurator Fiscal.
Area Commander for Orkney, Chief Inspector Matt Webb, said: “Police Scotland is absolutely committed to acting on information provided by the public to reduce the availability of drugs in our community and minimise the harm they cause. Thankfully a significant quantity of

In [ ]:

file = json.load(open(f'database_dump_drugs/{x}.json'))
print(file[0]['article_content'])

# Stanza Experimentation

In [3]:
def create_pipeline():
    # download the processors
    stanza.download('en', processors='tokenize,ner', logging_level='WARN')
    # loac the processors into a pipeline
    nlp = stanza.Pipeline('en', processors='tokenize,ner', logging_level='WARN')
    
    return nlp

In [ ]:
file = json.load(open('database_dump_drugs/0.json'))
print(file[0]['article_content'])

'An Ethiopian foreign national was arrested in eMzinoni for dealing in drugs at his tuck shop on Friday, 6 October.\nThis came after the police received a tip off from members of the Community Policing Forum (CPF) that the suspect was involved in the dealing of drugs.\nOfficers arrested the 27-year-old man.\nHe will appear in the Bethal Magistrate’s Court.\nPolice commended the CPF for their dedication and of fighting crime.\nFor free daily local news in the South, visit our sister newspapers Highvelder, Standerton Advertiser\nFor more news and interesting articles, like us on Facebook, and follow us on Twitter and Instagram'

In [4]:
pipeline = create_pipeline()
doc = pipeline(article)

In [ ]:
_ = 0
for sentence in doc.sentences:
     if _ == 0:
          print(sentence.ents)
          _ += 1
     else:
          # do nothing
          False

[{
  "text": "Ethiopian",
  "type": "NORP",
  "start_char": 3,
  "end_char": 12
}, {
  "text": "Friday",
  "type": "DATE",
  "start_char": 96,
  "end_char": 102
}, {
  "text": "6 October",
  "type": "DATE",
  "start_char": 104,
  "end_char": 113
}]
